streams/jvm/src/main/scala/zio/stream/platform.scala

```diff
@@ -213,6 +213,36 @@ trait ZStreamPlatformSpecificConstructors { self: ZStream.type =>
       .managed(is)
       .flatMap(fromInputStream(_, chunkSize))
 
+  /**
+   * Creates a stream from a callback that writes to [[java.io.OutputStream]].
+   * Note: the input stream will be closed after the `write` is done.
+   */
+  def fromOutputStream(
+    write: OutputStream => Unit,
+    chunkSize: Int = ZStream.DefaultChunkSize
+  ): ZStream[Blocking, Throwable, Byte] = {
+    def from(in: InputStream, out: OutputStream, err: Promise[Throwable, None.type]) = {
+      val readIn = fromInputStream(in, chunkSize).ensuring(ZIO.effectTotal(in.close()))
+      val writeOut = ZStream.fromEffect {
+        blocking
+          .effectBlocking(write(out))
```

Comment:
Would it make sense to use `blocking.effectBlockingInterrupt` to be able to interrupt this?

Reply:
Oh yeah probably, I didn't know about this 👍